DM-45482: Replace reference to setup.sh with envconfig and add how to change rubin-env (#687)
lsst-dm/dm_dev_guide: stack/conda.rst

Requests for additions to ``rubin-env`` should be made via RFC.
The DM-CCB must approve such requests.

.. _rubin-env-switching-versions:

Switching rubin-env versions
----------------------------

> There might be an instance where you might need to change the current version of rubin-env.

Reviewer: Drop the second "might".

> You can view the current available env `here <https://anaconda.org/conda-forge/rubin-env/files>`__

Reviewer: ```suggestion
You can view the current available env versions `on conda-forge <https://anaconda.org/conda-forge/rubin-env/files>`__
```

To see all available ``rubin-env`` in your terminal run the following:

.. code-block:: bash

   conda search -c nodefaults  -c conda-forge rubin-env

> This command will print out only production ready ``rubin-env``.

Reviewer: ```suggestion
This command will print out only production ready ``rubin-env`` versions.
```

To change the ``rubin-env``:

.. code-block:: bash

   git clone https://github.com/lsst/lsstsw
   cd lsstsw

>    LSST_CONDA_CHANNELS="nodefaults conda-forge"

Reviewer: This is the default so we don't need people to be setting it.

>    ./bin/deploy -v 8.0.0 # or any other rubin env

Reviewer: I'm sorry for not being clear. I know it can be run repeatedly, I was trying to ask if we've documented clearly on this page that you can run it as often as you want.

>    ./bin/deploy -v 8.0.0 # or any other rubin env

Author: Oh, okay. It is not documented clearly on this page, but is this the right page? When I joined, I assumed it was okay to rerun that command because I saw it being used in a few other places like https://arc.net/l/quote/atptszyd. I think it would be better to mention it in The LSST Software Build Tool section.

>    ./bin/deploy -v 8.0.0 # or any other rubin env

Reviewer: That link goes to a section that I don't think is still valid information now that we install EUPS via conda-forge. I think the whole section about setting eups versions can be removed because you get the eups version you get from conda-forge and if you want another version you use conda to change it. In the past eups was not on conda-forge and had to be installed separately by deploy.

Also, [that page](https://developer.lsst.io/stack/lsstsw.html#introduction) does indeed say that deploy can be run multiple times so my question is answered.

   source bin/envconfig -n lsst-scipipe-8.0.0
   rebuild lsst_apps # This could be any branch or product you want to use

> You might also want to switch to rubin-env that are still under development. To do so, add the label conda-forge/label/rubin-env_dev as follows:

Reviewer: I think this needs to be a subsection "Using developer rubin-env environments" so that it pops a bit and is easier to notice.

Also, we tend to prefer one sentence per line to make diffs easier.

.. code-block:: bash

   conda search -c nodefaults  -c conda-forge rubin-env -c conda-forge/label/rubin-env_dev

This command will print out all available ``rubin-env`` including ones that are still under development (rubin-env_dev).
To change the ``rubin-env``:

.. code-block:: bash

   git clone https://github.com/lsst/lsstsw
   cd lsstsw

>    LSST_CONDA_CHANNELS="nodefaults conda-forge/label/rubin-env_dev conda-forge" # adding conda-forge/label/rubin-env_dev allows to install rubin-env_dev

Reviewer: Move the comments to the line above otherwise they scroll off to the right and no-one will see them.

   ./bin/deploy -v 9.0.0dev # or any other rubin env
   source bin/envconfig -n lsst-scipipe-9.0.0dev

>    rebuild lsst_apps # This could be any branch or product you want to use

Reviewer: There should be a blank line after this.

Requests for additions to ``rubin-env`` should be made via RFC.
The DM-CCB must approve such requests.

.. _rubin-env-switching-versions:

Switching rubin-env versions
----------------------------

Occasionally, you may need to switch the version of ``rubin-env`` you are using.
Below, you'll find instructions on how to view available versions and switch between them.
You can view the current available env versions `on conda-forge <https://anaconda.org/conda-forge/rubin-env/files>`__.
To see all available ``rubin-env`` in your terminal run the following:

.. code-block:: bash

   conda search -c nodefaults  -c conda-forge rubin-env

This command will print out only production ready ``rubin-env`` versions.
To change the ``rubin-env``:

.. code-block:: bash

   conda deactivate # make sure you do not run this inside an existing conda environment
   ./bin/deploy -v 8.0.0 # or any other rubin env
   source bin/envconfig -n lsst-scipipe-8.0.0
   rebuild lsst_apps # This could be any branch or product you want to use

.. _rubin-env-dev-switching-versions:

Using developer rubin-env environments
--------------------------------------

> You might also want to switch to rubin-env that are still under development.

Reviewer: ```suggestion
You might also want to switch to ``rubin-env`` versions that are still under development.
```

> To do so, add the label conda-forge/label/rubin-env_dev as follows:

Reviewer: ```suggestion
To do so, add the label ``conda-forge/label/rubin-env_dev`` as follows:
```

.. code-block:: bash

   conda search -c nodefaults  -c conda-forge rubin-env -c conda-forge/label/rubin-env_dev

> This command will print out all available ``rubin-env`` including ones that are still under development (rubin-env_dev).

Reviewer: ```suggestion
This command will print out all available ``rubin-env`` including ones that are still under development (``rubin-env_dev``).
```

To change the ``rubin-env``:

.. code-block:: bash

>    conda deactivate # make sure you do not run this inside an existing conda environment

Reviewer: Maybe remind people here that it's better to start from a fresh shell whenever you run bin/deploy. I have found that very confusing things can happen if you don't do that (and we can remove the deactivate line if we do that).
